airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/migrations/SnowflakeAbMetaAndGenIdMigration.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.snowflake.migrations
+
+import io.airbyte.cdk.db.jdbc.JdbcDatabase
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.integrations.destination.jdbc.ColumnDefinition
+import io.airbyte.cdk.integrations.destination.jdbc.TableDefinition
+import io.airbyte.commons.json.Jsons
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationHandler
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationInitialStatus
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig
+import io.airbyte.integrations.base.destination.typing_deduping.migrators.Migration
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.util.LinkedHashMap
+
+private val log = KotlinLogging.logger {}
+
+class SnowflakeAbMetaAndGenIdMigration(private val database: JdbcDatabase) :
+    Migration<SnowflakeState> {
+    override fun migrateIfNecessary(
+        destinationHandler: DestinationHandler<SnowflakeState>,
+        stream: StreamConfig,
+        state: DestinationInitialStatus<SnowflakeState>
+    ): Migration.MigrationResult<SnowflakeState> {
+        if (!state.initialRawTableStatus.rawTableExists) {
+            // The raw table doesn't exist. No migration necessary. Update the state.
+            log.info {
+                "Skipping airbyte_meta/generation_id migration for ${stream.id.originalNamespace}.${stream.id.originalName} because the raw table doesn't exist"
+            }
+            return Migration.MigrationResult(state.destinationState, false)
+        }
+
+        val results =
+            database.queryJsons(
+                "SHOW COLUMNS IN TABLE \"${stream.id.rawNamespace}\".\"${stream.id.rawName}\""
+            )
+        val rawTableDefinition =
+            results
+                .groupBy { it.get("schema_name").asText()!! }
+                .mapValues { (_, v) ->
+                    v.groupBy { it.get("table_name").asText()!! }
+                        .mapValuesTo(LinkedHashMap()) { (_, v) ->
```

Comment:
nit: if you want to use an ordered map, probably should have a `sort` on ordinal before this?

Reply:
this particular query doesn't return ordinal column (unlike info_schema call).  `linkedHashMap()` explicit call is to make `TableDefinition` type happy. by default kotlin groupBy and other operations keeps the interation order intact by using an ordered map, its just that mapValuesTo uses super class `Map`